File: Examples/Examples/DataSource/Documentation Snippets/DataSourceDocsAuth.cs
using System;
using TelerikUI;
using Foundation;

namespace Examples
{
	// >> remote-auth-cs
	public class DataSourceDocsAuth : TKDataSource
	{
		public DataSourceDocsAuth ()
		{
		}
		public class MyDataSource: TKDataSource
		{
			[Export ("connection:didReceiveAuthenticationChallenge:")]
			public void ReceivedAuthenticationChallenge (NSUrlConnection connection, NSUrlAuthenticationChallenge challenge) 
			{
				if (challenge.PreviousFailureCount == 0) {
					var credential = new NSUrlCredential ("USER", "PASSWORD", NSUrlCredentialPersistence.ForSession);
					challenge.Sender.UseCredential (credential, challenge);
				}
				else {
					Console.WriteLine("previous authentication failure");
				}
			}
		}
	}
	// << remote-auth-cs
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/DataSourceDocsAuth.swift
//
//  DataSourceDocsAuth.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2016 Telerik. All rights reserved.
//

// >> remote-auth-swift
class DataSourceDocsAuth: TKDataSource {

    class MyDataSource: TKDataSource {
        
        override func connection(connection: NSURLConnection, didReceiveAuthenticationChallenge challenge: NSURLAuthenticationChallenge) {
            if challenge.previousFailureCount == 0 {
                let credential = NSURLCredential(user: "USER", password: "PASSWORD", persistence: NSURLCredentialPersistence.ForSession)
                challenge.sender?.useCredential(credential, forAuthenticationChallenge: challenge)
            }
            else {
                print("previous authentication failure")
            }
        }
    }
}
// << remote-auth-swift
